src/modules/theme/hocs/withTheme.js

```diff
@@ -0,0 +1,32 @@
+// @flow
+
+import * as React from 'react'
+import { wrapDisplayName } from 'recompose'
+
+import type { ThemeType } from '../constants/theme'
+import { brightColors } from '../constants/colors'
+import dimensions from '../constants/dimensions'
+import { arabicFonts, defaultFonts } from '../constants/fonts'
+
+type MapPropsToLanguageType<Props> = Props => string
+
+const ARABIC_LANGUAGES = ['ar', 'fa', 'ku']
+
+const withTheme = <Props> (mapPropsToLanguage?: MapPropsToLanguageType<Props>) =>
+  (Component: React.ComponentType<Props>): React.ComponentType<$Diff<Props, {| theme: ThemeType |}>> => {
+    return class extends React.Component<{}> {
```

Comment:
I'm not sure whether the flow type is correct here. Below you are accessing `this.props` which has the type `Props`. But in this line you are saying the prop type is `{}`. I think `Props` should be the right one. At least `{}` seems weird.

If you didn't have a look yet there is a quite good description: https://flow.org/en/docs/react/hoc/#toc-injecting-props